### PR TITLE
Random Battles updates

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
4 Attacks Scyther gets Choice Band instead of Boots
Solgaleo gets Weakness Policy again, and doesn't get double fire move
Various other movepool changes

also fixed some error in punishments using --fix i guess, couldn't actually send this commit to github without doing that.

### CODE REVIEW

File: server/punishments.ts [hunk 1]
```diff
@@ -1293,7 +1293,7 @@ export const Punishments = new class {
 
 		for (const curUser of affected) {
 			// ensure there aren't roombans so nothing gets mixed up
-			Punishments.roomUnban(room, (curUser as any).id || curUser);
+			Punishments.roomUnban(room, (curUser).id || curUser);
```
Comment (AnnikaCodes) [inline server/punishments.ts:1296]:
```suggestion
			Punishments.roomUnban(room, (curUser as any).id || curUser);
```

**Comment by AnnikaCodes** [comment]
I don't know what the Punishments thing is about, but I reverted it.

File: data/random-teams.ts [hunk 2]
```diff
@@ -971,6 +971,9 @@ export class RandomTeams {
 			// Fire Fang: Special case for Garchomp, which doesn't want Fire Fang w/o Swords Dance
 			const otherFireMoves = ['heatwave', 'overheat'].some(m => moves.has(m));
 			return {cull: (moves.has('fireblast') && counter.setupType !== 'Physical') || otherFireMoves};
+		case 'flareblitz':
+			// Special case for Solgaleo to prevent Flame Charge + Flare Blitz
+			return {cull: (species.id === 'solgaleo') && moves.has('flamecharge')};
```
Comment (AnnikaCodes) [inline data/random-teams.ts:976]:
```suggestion
			return {cull: species.id === 'solgaleo' && moves.has('flamecharge')};
```

**Comment by AnnikaCodes** [comment]
```
  1) [Gen 8] Random Battle
       should give 4 Attacks Scyther a Choice Band:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ 'Choice Scarf'
- 'Choice Band'
          ^
      + expected - actual

      -Choice Scarf
      +Choice Band
```
Your tests are failing; Scyther is getting Scarf sometimes.

**Comment by ACakeWearingAHat** [comment]
> 
> 
> ```
>   1) [Gen 8] Random Battle
>        should give 4 Attacks Scyther a Choice Band:
> 
>       AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
> + actual - expected
> 
> + 'Choice Scarf'
> - 'Choice Band'
>           ^
>       + expected - actual
> 
>       -Choice Scarf
>       +Choice Band
> ```
> 
> Your tests are failing; Scyther is getting Scarf sometimes.

could you fix that potentially

**Comment by AnnikaCodes** [comment]
Thanks!

**Pull request merged**